pkg/operation/botanist/component/extensions/dnsrecord/dnsrecord.go

```diff
@@ -275,3 +276,7 @@ func (c *dnsRecord) SetRecordType(recordType extensionsv1alpha1.DNSRecordType) {
 func (c *dnsRecord) SetValues(values []string) {
 	c.values.Values = values
 }
+
+func (c *dnsRecord) GetSecret() *corev1.Secret {
+	return c.secret
+}
```

Comment:
I believe the implementation of this method is wrong for its intended purpose. It will be used later to check if the name of the secret that **exists** contains `dnrecord-`. However, as currently implemented, it will return the name of the secret that has been initialized (but not necessarily already updated), and so it will return a `dnsrecord-` prefixed name also before the actual secret has been updated.

Reply:
Ah, thanks, that makes sense!